plugin/src/main/java/io/deephaven/plugin/type/ObjectTypeBase.java

```diff
@@ -9,8 +9,16 @@
 import java.io.OutputStream;
 
 public abstract class ObjectTypeBase extends PluginBase implements ObjectType {
```

Comment:
@devinrsmith can you remind me why this class exists at all, is it just to ensure that no one can accidentally replace the walk/accept method? Surely a default method could work for writeTo/writeCompatibleObjectTo?

Naming is a bit weird for "compatibleClientConnection" or whatnot, so I want to make sure I understand the intent (and to get Javadoc in here for the next time someone doesn't understand it).

Reply:
Yes; it's so nothing that extends from ObjectTypeBase can re-impl `writeTo` / `walk`. I personally shy away from default methods on interfaces when I can help it, but it's a personal preference, not a hard review point.